pkg/lib/kitfile/local-storage.go

```diff
@@ -89,64 +90,81 @@ func saveConfig(ctx context.Context, localRepo local.LocalRepo, kitfile *artifac
 	return desc, nil
 }
 
+// SaveKitfileLayers saves the kitfile layers with optional compression and returns a list of descriptors.
 func saveKitfileLayers(ctx context.Context, localRepo local.LocalRepo, kitfile *artifact.KitFile, ignore filesystem.IgnorePaths, compression string) ([]ocispec.Descriptor, error) {
 	var layers []ocispec.Descriptor
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(kitfile.Model.Parts)+len(kitfile.Code)+len(kitfile.DataSets)+len(kitfile.Docs))
+
+	processLayer := func(path string, mediaType constants.MediaType) {
+		defer wg.Done()
+		layer, err := saveContentLayer(ctx, localRepo, path, mediaType, ignore)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		mu.Lock()
+		layers = append(layers, layer)
+		mu.Unlock()
```

Comment:
I think this will cause some issues; currently the only way we have to match layer descriptors is through ordering, and this can result in a varying order for layers. 

We'll need to come up with an approach that ensures the `layers` list matches the order of elements in the kitfile (i.e. all datasets in the same order, etc.)

Reply:
What I mean here is that in the [unpack](https://github.com/jozu-ai/kitops/blob/4a7308af50bf3342a1c10d610889e1a78a156b19/pkg/cmd/unpack/unpack.go#L78-L80) process, iterate through each of the lists of items in the kitfile, matching indices (the second dataset we see in the manifest should be the second dataset in the kitfile)

With this approach, if, for example, we first pack a large dataset, then pack a smaller dataset, the smaller dataset may be added to `layers` before the large one.

Reply:
What we can do to solve this is to implement a simple layer indexing logic. A simple counter `layerIndex` can be incremented after each item is added to maintain the order. 
Each goroutine is given a layerIndex that corresponds to its position in the Kitfile, ensuring the order is preserved when layers are added to the layers slice.

Reply:
Good idea, this should work :+1: